src/main/java/task_manipulation/MarkAsDone.java

```diff
@@ -0,0 +1,43 @@
+package task_manipulation;
+
+import exceptions.DukeTaskInputException;
+import screen_output.Output_On_Screen;
+import task_classes.Task;
+
+import java.util.Vector;
+
+public class MarkAsDone {
+
+    /**
+     * The method to mark the task status as done
+     * If the is not format "done + Integer", system will add the entire input as Todo type task into the entire task list
+     *
+     * @param list the entire task list
+     * @param inputWords the string array of the user input
+     */
+    public static void markAsDone(Vector<Task> list, String[] inputWords) throws DukeTaskInputException {
+        if(list.isEmpty()){
+            throw new DukeTaskInputException(inputWords[0], "listIsEmtpy");
+        }
+
+        if (inputWords.length == 2) {
```

Comment:
Maybe can avoid arrowhead format by checking for error first?
```suggestion
        if (inputWords.length != 2) {
            //throw error 1
        } //error is checked and throw out before going to the next line
        if (!inputWords[1].matches("\\d+")){
           //throw error 2
        }
        //...
```